mlflow/sagemaker/__init__.py

```diff
@@ -1656,6 +1659,204 @@ def _does_model_exist(model_name, sage_client):
         return True if response else False
 
 
+class SageMakerPlugin(BaseDeploymentClient):
+    """
+    Initialize a deployment client for SageMaker
+
+    :param target_uri: A URI that has the value of ``sagemaker`` or ``sagemaker:/region_name``.
+                       The provided region name will be used as the default region for subsequent function calls.
+                       When no region name is provided, the default region will be set to `us-west-2`.
```

Comment:
Nice! Apologies for continuing to make things harder, but we recently merged https://github.com/mlflow/mlflow/pull/4923 to enable users to specify an IAM role to use when interfacing with SageMaker deployments, which makes it easier to manage deployments across accounts. It would be great to incorporate IAM roles into the target URI to support this use case as well.

Perhaps `sagemaker:/<region_name>/<iam_role_arn>`? Users may have to URL encode the role ARN if it contains `?` or `#`, but I think this is probably uncommon.

If an IAM role ARN is specified, the user must also specify the region. Let me know how this sounds.

Reply:
That sounds great!